students/merrix/session03/mymailroom.py

```diff
@@ -0,0 +1,101 @@
+import sys  # imports go at the top of the file
+from collections import defaultdict
+from operator import itemgetter
+
+donor_db = defaultdict(list)
+
+
+donor_db = {"William Gates":[653772, 12.17],
+            "Jeff Bezos": [877.33],
+            "Paul Allen" : [663.23, 43.87, 1.32],
+            "Mark Zuckerberg": [1663.23, 4300.87, 10432.0]
+           }
+
+prompt = "\n".join(("Welcome to James Dot Donations!",
+          "Please choose from below options:",
+          "1 - Send a Thank you",
+          "2 - Create a report",
+          "3 - Send Letter",
+          "4 - To Exit",
+          ">>> "))
+
+
+def view_donor_db():
+    print (donor_db.keys())
+    choose = input("enter a name from the list or a new name to donate: ").title()
+    if choose not in donor_db.keys():
+        add_donor_db(choose)
+    elif choose in donor_db.keys():
+        donation = int(input("Enter an amount to donate to %s " % choose)) 
+        donor_db[choose].append(int(donation))
+        print (donor_db)
+        print ('''
+        Thank you %s for your donation of $%i to this organization. 
+        We really appreciate your generousity and hope to see you again at our upcoming events.
+        Sincerely Yours
+        James Dot
+        ''' % (choose, donation))
+
+
+
+
+def add_donor_db(choose):
+    new_donor = input("Do you want to add %s to the donation list? yes or no " % choose).title()
+    if new_donor == "Yes":
+        amount = input("Enter an amount to donate for %s " % choose)
+        donor_db.update({new_donor:amount})
+        print (donor_db)
+        print ('''
+            Thank you %s for your donation of $%i to this organizat1ion. We really appreciate your generousity and hope to see you again at our upcoming events.
+            Sincerely Yours
+            James Dot
+            ''' % (new_donor, int(amount)))
```

Comment:
There are duplicates in your code, here in lines 47-52 and above in lines 31-37. I understand you had to handle the case of a new donor separately, because you have to initialize the new donor entry. However, once that's initialized, printing the letter is no different from the case of an existing donor, so this duplicate could have been avoided. It's OK for now, but please keep in mind that avoiding unnecessary duplicated code is very important in programming.